llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -2483,13 +2484,21 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
   using OriginCombiner = Combiner<false>;
 
   /// Propagate origin for arbitrary operation.
-  void setOriginForNaryOp(Instruction &I) {
+  ///
+  /// Optionally skips n trailing operands.
+  void setOriginForNaryOp(Instruction &I, unsigned int skipLastOperands = 0) {
```

Comment:
for this particular case skipping callee is unnecessary, it will be skipped in .Add()
as `getShadow` for that operand should be null.

Reply:
I prefer instead of extending this function we just to do 

```
    OriginCombiner OC(this, IRB);
    for (args)
      OC.Add(I.getOperand(i));
```

inside of

Reply:
> I prefer instead of extending this function we just to do
> 
> ```
>     OriginCombiner OC(this, IRB);
>     for (args)
>       OC.Add(I.getOperand(i));
> ```
> 
> inside of handleNEONVectorStoreIntrinsic

Done in https://github.com/llvm/llvm-project/pull/99360/commits/1970568ab31879094666b20d6f09b39682c20864